test/libsinsp_e2e/sys_call_test.cpp

```diff
@@ -1311,3 +1311,106 @@ TEST_F(sys_call_test, unshare_)
 	ASSERT_NO_FATAL_FAILURE({ event_capture::run(test, callback, filter); });
 	EXPECT_EQ(2, callnum);
 }
+
+TEST_F(sys_call_test, sendmsg_recvmsg_SCM_RIGHTS)
+{
+	int callnum = 0;
+	event_filter_t filter = [&](sinsp_evt* evt)
+	{
+		auto tinfo = evt->get_thread_info(true);
+		return tinfo->get_comm() == "libsinsp_e2e_te" && evt->get_type() == PPME_SOCKET_RECVMSG_X;
+	};
+	run_callback_t test = [&](concurrent_object_handle<sinsp> inspector_handle)
+	{
+		int server_sd, worker_sd, pair_sd[2];
+		int rc = socketpair(AF_UNIX, SOCK_DGRAM, 0, pair_sd);
+		ASSERT_GE(rc, 0);
+		server_sd = pair_sd[0];
+		worker_sd = pair_sd[1];
+
+		auto child = fork();
+		if (child == 0)
+		{
+			struct msghdr child_msg = {};
+			struct cmsghdr *cmsghdr;
+			struct iovec iov[1];
+			char buf[CMSG_SPACE(sizeof(int))], c;
+
+			iov[0].iov_base = &c;
+			iov[0].iov_len = sizeof(c);
+			memset(buf, 0x0d, sizeof(buf));
+			cmsghdr = (struct cmsghdr *)buf;
+			cmsghdr->cmsg_len = CMSG_LEN(sizeof(int));
+			cmsghdr->cmsg_level = SOL_SOCKET;
+			cmsghdr->cmsg_type = SCM_RIGHTS;
+			child_msg.msg_iov = iov;
+			child_msg.msg_iovlen = sizeof(iov) / sizeof(iov[0]);
+			child_msg.msg_control = cmsghdr;
+			child_msg.msg_controllen = CMSG_LEN(sizeof(int));
+			for (;;)
```

Comment:
Infinite loop to wait on the correct msg (ie: the `SCM_RIGHTS` one). Child should not see any other message though; perhaps we can remove the infinite loop?